File: MemeShareSwift/MemeShareSwift/model/UploadResponse.swift
//
//  UploadResponse.swift
//  MemeShareSwift
//
//  Created by Alperen Saraç on 31.08.2025.
//

import Foundation
struct UploadResponse: Decodable {
    let success: Bool
    let message: String?
    let mediaUrl: String?

    enum CodingKeys: String, CodingKey {
        case success, message
        case mediaUrl = "media_url"
    }
}

File: meme_share_flutter/lib/model/upload_response.dart
class UploadResponse {
  final bool success;
  final String message;
  final String mediaUrl;

  UploadResponse({
    required this.success,
    required this.message,
    required this.mediaUrl,
  });

  factory UploadResponse.fromJson(Map<String, dynamic> json) {
    return UploadResponse(
      success: json['success'] == true,
      message: (json['message'] ?? '').toString(),
      mediaUrl: (json['media_url'] ?? '').toString(),
    );
  }
}
